src/status_im/utils/keychain/core.cljs

```diff
@@ -129,14 +129,22 @@
   (comp validate
         string->js-array))
 
+(defonce generic-password (atom nil))
+
 (defn get-encryption-key []
-  (log/debug "initializing realm encryption key...")
-  (.. (.getGenericPassword rn/keychain)
-      (then
-       (fn [res]
-         (if res
-           (handle-found res)
-           (handle-not-found))))))
+  (log/debug "PERF" "initializing realm encryption key..." (.now js/Date))
+  (if @generic-password
+    (js/Promise.
+     (fn [on-success _]
+       (on-success @generic-password)))
+    (.. (.getGenericPassword rn/keychain)
+        (then
+         (fn [res]
+           (if res
+             (let [res' (handle-found res)]
+               (reset! generic-password res')
+               res')
```

Comment:
`reset!` will already return `res'`, so extra expression is not needed

Reply:
So the `let` expression might be simplified to just `(reset! generic-password (handle-found res`))`

Reply:
done